src/Dev/Deprecation.php

```diff
@@ -171,72 +173,79 @@ public static function set_enabled($enabled)
      */
     public static function notice($atVersion, $string = '', $scope = Deprecation::SCOPE_METHOD)
     {
-        if (!static::get_enabled()) {
+        if (static::$inside_notice) {
             return;
         }
+        static::$inside_notice = true;
+        try {
```

Comment:
The `try/catch` is only actually needed around the `user_error()` call section. Please make that change, which will make this diff a lot easier for me to parse and makes it a lot clearer what the purpose of that try is.

Reply:
Updated

Reply:
Actually, reverted back, I now remember why the try block needs to wrap everything

`static::get_enabled()` needs to be wrapped in the try block because it will later call some code that has `Deprecation::notice()` - specifically `BaseKernel::getEnvironment()`

Reply:
Please add a comment above the try block explaining this - otherwise I can see someone in the future going "Oh, we don't need this here" and causing a regression.

Reply:
Updated